src/configure/gen_c_primitives.ml

```diff
@@ -10,9 +10,119 @@ let header ="\
  * See the file LICENSE for details.
  *)
 
+(** Size and alignment of C primitives
+
+    This module defines the size and alignment of C primitives. By default, the
+    values are generated at compile time for the host architecture and compiler.
+
+    When the run-time architecture does not match the host architecture, a C
+    data model can be selected at compile-time or run-time using the environment
+    variable `CTYPES_DATA_MODEL`. Currently, only `CTYPES_DATA_MODEL=ILP32` is
+    supported, which can be used to execute programs in 32-bit WebAssembly.
+ *)
+
 open Ctypes_primitive_types
+
+module type CTYPES_PRIMITIVES = sig
+  val sizeof : 'a prim -> int
+  val alignment : 'a prim -> int
+  val pointer_size : int
+  val pointer_alignment : int
+end
+
 "
 
+let other_primitives = {|
+module Primitives_ILP32 = struct
+let char_size = 1
+let short_size = 2
+let int_size = 4
+let long_size = 4
+let llong_size = 8
+let size_t_size = 4
+let pointer_size = 4
+let pointer_alignment = 4
+let sizeof : type a. a prim -> int = function
+  | Char -> char_size
+  | Schar -> char_size
+  | Uchar -> char_size
+  | Bool -> 1
+  | Short -> short_size
+  | Int -> int_size
+  | Long -> long_size
+  | Llong -> llong_size
+  | Ushort -> short_size
+  | Sint -> int_size
+  | Uint -> int_size
+  | Ulong -> long_size
+  | Ullong -> llong_size
+  | Size_t -> size_t_size
+  | Int8_t -> 1
+  | Int16_t -> 2
+  | Int32_t -> 4
+  | Int64_t -> 8
+  | Uint8_t -> 1
+  | Uint16_t -> 2
+  | Uint32_t -> 4
+  | Uint64_t -> 8
+  | Float -> 4
+  | Double -> 8
+  | LDouble -> 16
+  | Complex32 -> 8
+  | Complex64 -> 16
+  | Complexld -> 32
+  | Nativeint -> 4
+  | Camlint -> 4
+let alignment : type a. a prim -> int = function
+  | Char -> 1
+  | Schar -> 1
+  | Uchar -> 1
+  | Bool -> 1
+  | Short -> 2
+  | Int -> 4
+  | Long -> 4
+  | Llong -> 8
```

Comment:
I think this is only 4 (remember, the data bus is 32 bits, so it is no problem for the CPU to split a 64 bit lookup into two 32 bit lookups at 4-aligned addresses).

Reply:
Hmmm, wasm has an 8-alignment for long long. Compare e.g. with this page: https://developer.apple.com/library/archive/documentation/Darwin/Conceptual/64bitPorting/transition/transition.html. This detail seems to be not fully standardized.